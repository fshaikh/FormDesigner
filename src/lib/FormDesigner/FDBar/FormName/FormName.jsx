/**
 * React specific imports
 */
import React from 'react';

/**
 * Import material-ui components
 */
import TextField from '@material-ui/core/TextField';

/**
 * Library specific imports
 */
import FormDesignerContext from '../../Store/FormDesignerContext';
import * as Actions from '../../Actions/Actions';

/**
 * Presentational component for representing Form Name UI
 * Uses TextField component of Material UI for rendering the text field
 * Consumes Context to read form definition and dispatch action when form name value changes
 */
const FormName = () => {
    return (
        <FormDesignerContext.Consumer>
            { (context) => { 
                return <TextField defaultValue={context.formDefinition.formName}
                              placeholder={context.strings.FormNamePlaceholder}
                              onChange={(event) => context.eventEmitter.dispatch(Actions.FormNameChangeAction(event.target.value))}     
                        />  
                }
            }
        </FormDesignerContext.Consumer>
    );
};

export default FormName;