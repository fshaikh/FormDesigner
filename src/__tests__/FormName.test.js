import React from 'react';
import renderer from 'react-test-renderer';
import getElementWithContext from 'react-test-context-provider'

import * as Core from './core';
import FormName from '../lib/FormDesigner/FDBar/FormName/FormName';


describe('Form Name component', () => {
    describe('snapshot tests', () => {
        it('should render correctly', () => {
            const Provider = Core.getContextProvider()
            const component = renderer.create(
                    <Provider value={Core.getContextDefault()}><FormName /></Provider>
                ).toJSON();
            expect(component).toMatchSnapshot()
        });
    });
});