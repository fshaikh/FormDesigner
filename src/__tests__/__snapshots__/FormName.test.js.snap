// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Name component snapshot tests should render correctly 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-01"
>
  <div
    className="MuiInput-root-05 MuiInput-formControl-06 MuiInput-underline-09"
  >
    <input
      aria-invalid={false}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-013"
      defaultValue="New Form"
      disabled={false}
      id={undefined}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Enter form name here"
      readOnly={undefined}
      required={false}
      rows={undefined}
      type="text"
      value={undefined}
    />
  </div>
</div>
`;
